torchgen/dest/register_dispatch_key.py

```diff
@@ -609,7 +609,7 @@ def gen_class(
         elif self.backend_index.dispatch_key == DispatchKey.CompositeExplicitAutograd:
             guard_field = "c10::OptionalDeviceGuard guard_;"
         elif self.backend_index.dispatch_key == DispatchKey.MPS:
-            guard_field = "c10::OptionalMPSGuard guard_;"
+            guard_field = "c10::OptionalDeviceGuard guard_;"
```

Comment:
@ezyang , this part I had to revert the OptionalMPSGuard to OptionalDeviceGuard. As the MPSGuard was missing from c10. We will have to add a struct in c10 specifically for mps to make it work.

Reply:
@albanD FYI, I will add a TODO there.